joblib/test/test_parallel.py

```diff
@@ -248,6 +250,23 @@ def g(x):
                   (delayed(g)(x) for x in range(10)))
 
 
+def test_parallel_timeout_success():
+    # Check that timeout isn't thrown when function is fast enough
+    nose.tools.assert_equal(
+        10,
+        len(Parallel(n_jobs=2, timeout=10)
+            (delayed(sleep)(0.001) for x in range(10))))
+
+
+def test_parallel_timeout_fail():
+    # Check that timeout properly fails when function is too slow
+    nose.tools.assert_raises(
+        TimeoutError,
+        Parallel(n_jobs=2, timeout=0.01),
+        (delayed(sleep)(10) for x in range(10))
+    )
```

Comment:
Could you please extend those test to check that this also work with `backend="threading"`?

```python
for backend in ['multiprocessing', 'threading']:
    # do tests with Parallel(n_jobs=2, backend=backend, timeout=0.01)
```